src/components/constraints/ConstraintForm.svelte

```diff
@@ -243,11 +243,11 @@
         <button
           class="st-button {saveButtonClass} ellipsis"
           disabled={!saveButtonEnabled}
-          on:click={saveConstraint}
           use:permissionHandler={{
             hasPermission,
             permissionError,
           }}
+          on:click={saveConstraint}
```

Comment:
Why the reordering?

Reply:
So there's a difference in how Firefox and Chrome observe the `useCapture` flag in [event listeners](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener). In Chrome, if `useCapture` is passed in as true to an event listener, its callback seems to be executed first regardless of when the event was added to the element. However in Firefox, the event callbacks are invoked in order of when they were added to the element.

Here is an example:
https://codepen.io/pen/XWyvBWM

I'm reordering in order to let the `permissionHandler` add its click event listener first.

Reply:
Oh man okay. So anytime we use `permissionHandler` we need to account for this? Maybe we should add a comment in the template just to note this.